python/mcap-ros2-support/mcap_ros2/cdr.py

```diff
@@ -0,0 +1,162 @@
+        self.offset += 2
+        return value
+
+    def int32(self) -> int:
+        self.__align(4)
+        fmt = "<i" if self.little_endian else ">i"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 4
+        return value
+
+    def uint32(self) -> int:
+        self.__align(4)
+        fmt = "<I" if self.little_endian else ">I"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 4
+        return value
+
+    def int64(self) -> int:
+        self.__align(8)
+        fmt = "<q" if self.little_endian else ">q"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 8
+        return value
+
+    def uint64(self) -> int:
+        self.__align(8)
+        fmt = "<Q" if self.little_endian else ">Q"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 8
+        return value
+
+    def uint16BE(self) -> int:
+        self.__align(2)
+        value = struct.unpack_from(">H", self.data, self.offset)[0]
+        self.offset += 2
+        return value
+
+    def uint32BE(self) -> int:
+        self.__align(4)
+        value = struct.unpack_from(">I", self.data, self.offset)[0]
+        self.offset += 4
+        return value
+
+    def uint64BE(self) -> int:
+        self.__align(8)
+        value = struct.unpack_from(">Q", self.data, self.offset)[0]
+        self.offset += 8
+        return value
+
+    def float32(self) -> float:
+        self.__align(4)
+        fmt = "<f" if self.little_endian else ">f"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 4
+        return value
+
+    def float64(self) -> float:
+        self.__align(8)
+        fmt = "<d" if self.little_endian else ">d"
+        value = struct.unpack_from(fmt, self.data, self.offset)[0]
+        self.offset += 8
+        return value
+
+    def string(self) -> str:
+        length = self.uint32()
+        if length <= 1:
+            self.offset += length
+            return ""
+        return self.string_raw(length - 1)
+
+    def string_raw(self, length: int) -> str:
+        data = self.uint8_array(length)
+        value = data.decode("utf-8")
+        return value
+
+    def sequence_length(self) -> int:
+        return self.uint32()
+
+    def uint8_array(self, length: int) -> bytes:
+        data = self.data[self.offset : self.offset + length]
+        self.offset += length
+        return data
+
+    def seek(self, relative_offset: int):
+        new_offset = self.offset + relative_offset
+        if new_offset < 4 or new_offset >= len(self.data):
+            raise Exception(
+                f"seek({relative_offset}) failed, {new_offset} is outside the data range"
+            )
+        self.offset = new_offset
+
+    def seek_to(self, offset: int):
+        if offset < 4 or offset >= len(self.data):
+            raise Exception(
```

Comment:
avoid raising bare Exceptions, RuntimeError is appropriate here
```suggestion
            raise RuntimeError(
```